scalding-core/src/main/scala/com/twitter/scalding/TupleConverter.scala

```diff
@@ -67,6 +100,23 @@ object TupleConverter extends GeneratedTupleConverters {
   def arity[T](implicit tc: TupleConverter[T]): Int = tc.arity
   def of[T](implicit tc: TupleConverter[T]): TupleConverter[T] = tc
 
+  import shapeless.ops.hlist._
+
+  implicit def hListConverter[H, T <: HList, N <: Nat]
+    (implicit
+     g: TupleGetter[H],
+     len: Length.Aux[H :: T, N],
+     toIntN : ToInt[N],
+     fl : FromTupleEntry[H :: T, Nat._0]): TupleConverter[H :: T] =
+      new TupleConverter[H :: T] {
+        override def apply(te: TupleEntry): H :: T = {
+          val l : Option[H :: T] = fl(te)
+          l.get
```

Comment:
So, TupleConverter and TupleSetter are the places where types are added and removed before handing to cascading. It can't be type-safe at that point because cascading is not type-safe. There is really nothing you can do but call .get. That said, in a well typed scalding flow, that .get can never throw, so ultimately it would be safe (even though using the TupleConverter on its own, would not be safe).